pkg/minikube/tunnel/route_windows.go

```diff
@@ -0,0 +1,158 @@
+	if e != nil {
+		return e
+	}
+	if exists {
+		return nil
+	}
+
+	serviceCIDR := r.config.RoutedCIDR.String()
+	destinationIP := r.config.RoutedCIDR.IP.String()
+	destinationMask := fmt.Sprintf("%d.%d.%d.%d",
+		r.config.RoutedCIDR.Mask[0],
+		r.config.RoutedCIDR.Mask[1],
+		r.config.RoutedCIDR.Mask[2],
+		r.config.RoutedCIDR.Mask[3])
+
+	gatewayIP := r.config.TargetGateway.String()
+
+	logrus.Infof("Adding route for CIDR %s to gateway %s", serviceCIDR, gatewayIP)
+	command := exec.Command("route", "ADD", destinationIP, "MASK", destinationMask, gatewayIP)
+	logrus.Infof("About to run command: %s", command.Args)
+	stdInAndOut, e := command.CombinedOutput()
+	message := fmt.Sprintf("%s", stdInAndOut)
+	if message != " OK!\r\n" {
+		return fmt.Errorf("error adding route: %s, %d", message, len(strings.Split(message, "\n")))
+	}
+	logrus.Infof("%s", stdInAndOut)
+	if e != nil {
+		logrus.Errorf("error adding route: %s, %d", message, len(strings.Split(message, "\n")))
+		return e
+	}
+	return nil
+}
+
+func checkRoute(cidr *net.IPNet, gateway net.IP) (bool, error) {
+	stdInAndOut, e := exec.Command("route", "print", "-4").CombinedOutput()
+	if e != nil {
+		return false, e
+	}
+	routeTableString := fmt.Sprintf("%s", stdInAndOut)
+	return checkRouteTable(cidr, gateway, routeTableString)
+}
+
+func checkRouteTable(cidr *net.IPNet, gateway net.IP, routeTableString string) (bool, error) {
+	routeTable := strings.Split(routeTableString, "\n")
+	skip := true
+	exactMatch := false
+	collision := ""
+	for _, line := range routeTable {
+		//after first line of header we can start consuming
+		if strings.HasPrefix(line, "Network Destination") {
+			skip = false
+			continue
+		}
+
+		fields := strings.Fields(line)
+		//don't care about the 0.0.0.0 routes
+		if skip || len(fields) == 0 || len(fields) > 0 && (fields[0] == "default" || fields[0] == "0.0.0.0") {
+			continue
+		}
+		if len(fields) > 2 {
+			dstCIDRIP := net.ParseIP(fields[0])
+			dstCIDRMask := fields[1]
+			dstMaskIP := net.ParseIP(dstCIDRMask)
+			gatewayIP := fields[2]
+			dstCIDR := &net.IPNet{
+				IP:   dstCIDRIP,
+				Mask: net.IPMask(dstMaskIP.To4()),
+			}
+			if dstCIDR.String() == cidr.String() {
+				if gatewayIP == gateway.String() {
+					exactMatch = true
+				} else {
+					collision = line
+				}
+			} else if dstCIDR != nil {
+				if dstCIDR.Contains(cidr.IP) || cidr.Contains(dstCIDRIP) {
+					logrus.Warningf("overlapping CIDR (%s) detected in routing table with minikube tunnel (%s). It is advisable to remove this rule. Run: sudo route -n delete %s", dstCIDR.String(), cidr, dstCIDR.String())
+				}
+			} else {
+				logrus.Errorf("can't parse CIDR from routing table: %s", dstCIDR)
+			}
+		}
+	}
+
+	if exactMatch {
+		return true, nil
+	}
+
+	if len(collision) > 0 {
+		return false, fmt.Errorf("conflicting rule in routing table: %s", collision)
+	}
+
+	return false, nil
+}
+
+func (r *OSRouter) Cleanup() error {
```

Comment:
if minikube tunnel crashes or exits unexepectedly, is there a way for users to manually run this operation, e.g. minikube tunnel --cleanup

Reply:
not yet, I do like the idea. Would this belong in this PR or can be done in a separate PR?

Reply:
I added this feature